### ISSUE TITLE
laterpay/scripts/lp-balance.php must be replaced

### ISSUE THREAD

**Issue by drdla** [laterpay/laterpay-wordpress-plugin#69]
I found this script embedded on Richard's blog. Unfortunately it will never work correctly.

The IFrame API has some peculiar properties:
1. LaterPay's IFrame JavaScript `Y.LaterPay.IFrame` _must create_ the IFrame containing the balance, as it spawns it for easyXDM. There is _no_ way for connecting to an existing IFrame. So for the balance iframe on wordpress sites to become realtime, we need to embed the following script in the site:
   
   ``` javascript
   var balance_iframe = new Y.LaterPay.IFrame(Y.one('#balancecontainer'),
       '{{ invoicebalance_iframe_url }}',
       {
           width: "150",
           height: "30",
           scrolling: "no",
           frameborder: "0",
           style: "float: right"
       }
   );
   ```
2. After the IFrame has been spawned by `Y.LaterPay.IFrame`, LaterPay runs JavaScript inside it that connects to the outside page. Unfortunately, the HTML5 `postMessage` API _prohibits any URL changes_ inside the IFrame and breaks the connection to the outside page on a redirect. This is part of the browser security model. 
   
    The `lp-balance.php` script currently uses a redirect and breaks that connection, as evidenced by the JavaScript error `TypeError: url is undefined - url = url.replace(reDoubleSlash, "$1/");` that currently occurs on every page in `easyXDM`. 
   
    On the LaterPay dialogs we solved that particular problem by embedding a second IFrame inside the first IFrame which provides a bridge object to the inner IFrame. That way the outer IFrame's URL never changes. However, that approach only works when both, the outer and inner IFrame, load from the exact same domain, thanks to the browser security model.

**tl;dr:**
lp-balance.php can never work. The right way to solve this is to check for live mode and either render a `web.sandbox.laterpaytest.net` URL or a `web.laterpay.net` URL into a `<script>` tag in the footer of the page. If we rather incur another HTTP request you can also make the same check in a PHP script that returns the JavaScript directly and call it from a `<script src="..."></script>` tag.
